src/components/Table/GridTable.tsx

```diff
@@ -274,6 +290,111 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
   const expandedColumnIds: string[] = useComputed(() => tableState.expandedColumnIds, [tableState]);
   const columnSizes = useSetupColumnSizes(style, columns, resizeTarget ?? resizeRef, expandedColumnIds);
 
+  function isCursorBelowMidpoint(target: HTMLElement, clientY: number) {
+    const style = window.getComputedStyle(target);
+    const rect = target.getBoundingClientRect();
+
+    const pt = parseInt(style.getPropertyValue("padding-top")) / 2;
+    const pb = parseInt(style.getPropertyValue("padding-bottom"));
+
+    return clientY > rect.top + pt + (rect.height - pb) / 2;
+  }
+
+  // allows us to unset children and grandchildren, etc.
+  function recursiveSetDraggedOver(rows: GridDataRow<R>[], draggedOver: DraggedOver) {
+    rows.forEach((r) => {
+      tableState.maybeSetRowDraggedOver(r.id, draggedOver);
+      if (r.children) {
+        recursiveSetDraggedOver(r.children, draggedOver);
+      }
+    });
+  }
+
+  function onDragStart(row: GridDataRow<R>, evt: DragEventType) {
+    if (!row.draggable || !droppedCallback) {
+      return;
+    }
+
+    evt.dataTransfer.effectAllowed = "move";
+    evt.dataTransfer.setData("text/plain", JSON.stringify({ row }));
```

Comment:
Thought: The row might be a little big, although maybe it doesn't matter? I'd be tempted to just do like `setdata("text/plain", row.id)` because the row ids should be unique.